cpp/oneapi/dal/table/detail/csr.hpp

```diff
@@ -0,0 +1,179 @@
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/detail/array_utils.hpp"
+#include "oneapi/dal/table/detail/sparse_access_iface.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+class ONEDAL_EXPORT csr_table : public table {
+    friend detail::pimpl_accessor;
+
+public:
+    /// Returns the unique id of ``csr_table`` class.
+    static std::int64_t kind();
+
+    /// Creates a new ``csr_table`` instance with zero number of rows and columns.
+    /// The :expr:`kind` is set to``csr_table::kind()``.
+    /// All the properties should be set to default values (see the Properties section).
+    csr_table();
+
+    /// Creates a new ``csr_table`` instance from externally-defined data blocks.
+    /// Table object owns the data, row indices and column indices pointers.
+    ///
+    /// @tparam Data          The type of elements in the data block that will be stored into the table.
+    ///                       The :literal:`Data` type should be at least :expr:`float`, :expr:`double` or :expr:`std::int32_t`.
+    /// @tparam DataDeleter   The type of a deleter called on ``data`` when
+    ///                       the last table that refers it is out of the scope.
+    /// @tparam ColumnDeleter The type of a deleter called on ``column_indices`` when
+    ///                       the last table that refers it is out of the scope.
+    /// @tparam RowDeleter    The type of a deleter called on ``row_indices`` when
+    ///                       the last table that refers it is out of the scope.
+    ///
+    /// @param data           The pointer to values in the CSR layout.
+    /// @param column_indices The pointer to column indices in the CSR layout.
+    /// @param row_indices    The pointer to row indices in the CSR layout.
+    /// @param column_count   The number of columns in the corresponding dense table.
+    /// @param row_count      The number of rows in the corresponding dense table.
+    /// @param data_deleter   The deleter that is called on the ``data`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param column_deleter The deleter that is called on the ``column_indices`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param row_deleter    The deleter that is called on the ``row_indices`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param indexing       The indexing scheme used to access data in the CSR layout. Support only :literal:`csr_indexing::one_based`.
+    template <typename Data, typename DataDeleter, typename ColumnDeleter, typename RowDeleter>
+    csr_table(const Data* data,
+              const std::int64_t* column_indices,
+              const std::int64_t* row_indices,
+              std::int64_t column_count,
```

Comment:
inconsistent with homogen table. swap columns and row counts

Reply:
It was fixed